### PR TITLE
Interface Issue

### PR DESCRIPTION
**Pull request by swalker2m**
Works around the GraphQL `interface` path limitation by avoiding `interface`s.

### CODE REVIEW

**Review by toddburnside — APPROVED** [review]
LGTM. I think this approach works best for our API given our constraints. Thanks for tackling this.

File: modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql [hunk 1]
```diff
@@ -9395,6 +9094,17 @@ interface StepRecord {
...
+  GMOS North instrument configuration for this step, if any.  This will be null
+  unless the `instrument` discriminator is "GMOS_NORTH".
+  """
+  gmosNorth: GmosNorthDynamic!
+
+  """
+  GMOS South instrument configuration for this step, if any.  This will be null
+  unless the `instrument` discriminator is "GMOS_SOUTH".
+  """
+  gmosSouth: GmosSouthDynamic!
```
Comment (swalker2m) [inline modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql:9107]: Oops, these should be optional now.  Not sure how that worked.

File: modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql [hunk 2]
```diff
@@ -10110,6 +9820,16 @@ interface Visit {
...
 
+  """
+  GmosNorth static instrument configuration, for GMOS North visits.
+  """
+  gmosNorth: GmosNorthStatic!
+
+  """
+  GmosSouth static instrument configuration, for GMOS South visits.
+  """
+  gmosSouth: GmosSouthStatic!
```
Comment (swalker2m) [inline modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql:9831]: Make optional.

**Review by tpolecat — APPROVED** [review]
This all kind of sucks but it's better to discover it now I guess. 👍🏻

**Pull request merged**